framework/src/modules/interoperability/sidechain/commands/mainchain_registration.ts

```diff
@@ -193,11 +201,44 @@ export class MainchainRegistrationCommand extends BaseInteroperabilityCommand {
 		await outboxRootSubstore.set(context, MAINCHAIN_ID_BUFFER, { root: EMPTY_HASH });
 
 		const ownChainAccountSubstore = this.stores.get(OwnChainAccountStore);
-		await ownChainAccountSubstore.set(context, utils.intToBuffer(0, 4), {
+		await ownChainAccountSubstore.set(context, EMPTY_BYTES, {
 			name: ownName,
 			id: ownChainID,
 			nonce: BigInt(0),
 		});
+		this.events
+			.get(ChainAccountUpdatedEvent)
+			.log(methodContext, MAINCHAIN_ID_BUFFER, mainchainAccount);
+
+		const encodedParams = codec.encode(registrationCCMParamsSchema, {
```

Comment:
you could get this by `context.transaction.params`

Reply:
I am not sure I understand how. Can you please elaborate ?

Reply:
ah sorry, i think it was misunderstanding.
I thought `registrationCCMParamsSchema` is the same schema as the mainchain registration